### PR TITLE
implement the abort_on_failure functionality

### LINKED ISSUE #667: abort_on_failure never used?
We have the runtime parameter abort_on_failure, but it does not seem that we use it any more.

In Castro, we take pains to override this in certain situations, so that Castro logic might be doing nothing.

### PR DESCRIPTION
**Pull request by zingale**
we lost this in the F90 -> C++ port
this closes #667

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2021-06-20-001/index.html

**Comment by zingale** [comment]
we actually had a failure in the suite that we were not detecting.

**Review by maxpkatz — COMMENTED** [review]
Need to wrap the error in ifndef AMREX_USE_GPU.

**Comment by zingale** [comment]
done

File: integration/integrator_sdc.H [hunk 1]
```diff
@@ -7,6 +7,19 @@ AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
 void integrator (burn_t& state, Real dt)
 {
     actual_integrator(state, dt);
+
+#ifndef NSE_TABLE
+    // optionally abort here if we don't have retries in the driver.
+    // Note, for NSE we get a second chance if the state entered NSE,
+    // in burner()
+    if (!state.success && abort_on_failure) {
+        std::cout << "failure: \n";
```
Comment (maxpkatz) [inline integration/integrator_sdc.H:16]: The cout also needs to be in the ifndef

**Comment by zingale** [comment]
all tests pass: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2021-06-20-002/index.html

File: integration/integrator_sdc.H (hunk 1 above)
Comment (zingale) [reply integration/integrator_sdc.H:17]: fixed

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**